silverstripe/silverstripe-admin
# FIX Focus on first tab with field that failed validation

**Update: we're unlikely to merge this, because even though it's a regression from previous behaviour, we'll probably enhance this and put alert icons on tabs with invalid form fields, instead of tab switching**

---

Partial fix for https://github.com/silverstripe/silverstripe-cms/issues/2081

Based on Robbie noticing this https://github.com/silverstripe/silverstripe-cms/issues/2081#issuecomment-515889742

Seems like this is a regression of functionality that used to work.  At some point we must have change the classes used for tabs and validation messages, as well as possibly how tabs are structured.  Likely while going from SS3 to SS4.

Testing setup:

**Page.php**

```php
<?php

namespace {

    use SilverStripe\CMS\Model\SiteTree;
    use SilverStripe\Forms\RequiredFields;
    use SilverStripe\Forms\TextField;

    class Page extends SiteTree
    {
        private static $db = [
            'ABC' => 'Varchar',
            'DEF' => 'Varchar',
            'XYZ' => 'Varchar',
            'S22' => 'Varchar',
        ];

        private static $has_one = [];

        public function getCMSFields()
        {
            $fields = parent::getCMSFields();
            $fields->addFieldToTab("Root.Second", TextField::create("ABC"));

            // used to test javascript scroll into view
            $fields->addFieldToTab("Root.Third",TextField::create("null1"));
            $fields->addFieldToTab("Root.Third",TextField::create("null2"));
            $fields->addFieldToTab("Root.Third",TextField::create("null3"));
            $fields->addFieldToTab("Root.Third",TextField::create("null4"));
            $fields->addFieldToTab("Root.Third",TextField::create("null5"));
            $fields->addFieldToTab("Root.Third",TextField::create("null6"));
            $fields->addFieldToTab("Root.Third",TextField::create("null7"));
            $fields->addFieldToTab("Root.Third",TextField::create("null8"));
            $fields->addFieldToTab("Root.Third",TextField::create("null9"));
            $fields->addFieldToTab("Root.Third",TextField::create("null10"));
            $fields->addFieldToTab("Root.Third",TextField::create("null11"));
            $fields->addFieldToTab("Root.Third",TextField::create("null12"));
            $fields->addFieldToTab("Root.Third",TextField::create("null13"));
            $fields->addFieldToTab("Root.Third",TextField::create("null14"));

            $fields->addFieldToTab("Root.Third", TextField::create("DEF"));
            
            $fields->addFieldToTab("Root.Fourth", TextField::create("XYZ"));
            return $fields;
        }

        public function getSettingsFields()
        {
            $fields = parent::getSettingsFields();
            $fields->addFieldToTab("Root.Settings", TextField::create('S22'));
            return $fields;
        }

        public function getCMSValidator() {
            return new RequiredFields(array('DEF', 'XYZ', 'S22'));
        }
    }
}
```

1) Click Save without filling in any fields
2) It should focus on the Tab "Third" and you should see a `"DEF" is required` validation message.
3) Fill the DEF field in, click save
4) You should now be focused on the Tab "Fourth" and see a `"XYZ" is required` validation message.
